### PR TITLE
feat: API for list of OpenCallApplications for private area

### PR DESCRIPTION
**Pull request by martintomas**
New API endpoint (`GET /api/v1/account/open_call_applications`) which returns list of open call applications for:
 - investor: applications for investor open calls
 - project_developer: applications for PD projects

Data returned by this endpoint are for both type of accounts same, therefore FE is in full control of what is shown (works same way as other endpoints). Endpoint supports `includes` param so it is possible to obtain details of investor/project/project_developer/open_call at any point <-- required for example for details.

Searching works slightly differently for each account type, so I have split it and different ransack query is used for Investor and PD.

## Testing instructions

New endpoint is again part of rswag documentation. Login as PD and check that you get list of applications which you have created. Login as Investor and check that you see all applications for your all open calls.

## Tracking

https://vizzuality.atlassian.net/browse/LET-890
https://vizzuality.atlassian.net/browse/LET-923
https://vizzuality.atlassian.net/browse/LET-895
https://vizzuality.atlassian.net/browse/LET-972
https://vizzuality.atlassian.net/browse/LET-924
https://vizzuality.atlassian.net/browse/LET-971

### CODE REVIEW

**Review by tsubik — COMMENTED** [review]
At some point it would be worth it to update seeds to contain more test data, some project developers and investors with open calls applications, as creating that manually is very time-consuming.

File: backend/app/controllers/api/v1/accounts/open_call_applications_controller.rb [hunk 1]
```diff
@@ -5,6 +5,20 @@ class OpenCallApplicationsController < BaseController
         around_action(only: %i[create]) { |_, action| set_locale(language: current_user&.account&.language, &action) }
         load_and_authorize_resource
 
+        def index
+          @open_call_applications = @open_call_applications.includes(open_call: {investor: :account}, project: {project_developer: :account})
+          @open_call_applications = @open_call_applications.where project_id: filter_params[:project_id] if filter_params[:project_id].present?
```
Comment (tsubik) [inline backend/app/controllers/api/v1/accounts/open_call_applications_controller.rb:10]: I did not find a requirement for this and to filter by open_call_id but I trust this is needed :)

Comment (tsubik) [reply backend/app/controllers/api/v1/accounts/open_call_applications_controller.rb:10]: Ahh, I guess this is for details endpoint, to get only those for project/open call.

Comment (tsubik) [reply backend/app/controllers/api/v1/accounts/open_call_applications_controller.rb:10]: or should we add a separate show endpoint for details?

**Comment by martintomas** [comment]
> At some point it would be worth it to update seeds to contain more test data, some project developers and investors with open calls applications, as creating that manually is very time-consuming.

I can add `open call application` creation to seeding, np

**Review by tsubik — APPROVED** [review]
Looks great! :+1:

**Pull request merged**